### PR TITLE
chore: move basic bundle plugin to core

### PR DESCRIPTION
**Pull request by NoemiRozpara**

### CODE REVIEW

File: packages/haul-core/src/preset/makeConfigFactory.ts [hunk 1]
```diff
@@ -27,6 +26,7 @@ import applySingleBundleTweaks from './utils/applySingleBundleTweaks';
 import applyMultiBundleTweaks from './utils/applyMultiBundleTweaks';
 import LooseModeWebpackPlugin from '../webpack/plugins/LooseModeWebpackPlugin';
 import InitCoreDllPlugin from '../webpack/plugins/InitCoreDllPlugin';
+import WebpackBasicBundlePlugin from '../webpack/plugins/WebpackBasicBundlePlugin';
```
Comment (thymikee) [inline packages/haul-core/src/preset/makeConfigFactory.ts:29]: Can we come up with consistent naming?
We now have: 
- LooseModeWebpackPlugin
- WebpackBasicBundlePlugin

I think the easiest would be to go from `WebpackBasicBundlePlugin` to `BundleWebpackBasicPlugin`

File: packages/haul-basic-bundle-webpack-plugin/package.json [hunk 2]
```diff
@@ -1,21 +0,0 @@
...
-  "bugs": "https://github.com/callstack/haul/issues",
-  "peerDependencies": {
-    "webpack": "4.x",
-    "webpack-sources": "1.x"
-  },
-  "dependencies": {
-    "concat-with-sourcemaps": "^1.1.0"
-  },
-  "gitHead": "c41becc82dfeafb07b642ebed8f082775647ec3f"
-}
```
Comment (zamotany) [inline packages/haul-basic-bundle-webpack-plugin/package.json:21]: We still need that file to publish it to NPM and then deprecate that.

File: packages/haul-basic-bundle-webpack-plugin/index.ts [hunk 3]
```diff
@@ -0,0 +1 @@
+export { default } from '../haul-core/src/webpack/plugins/WebpackBasicBundlePlugin';
```
Comment (zamotany) [inline packages/haul-basic-bundle-webpack-plugin/index.ts:1]: This should be `index.js` file and it should be using CJS:
```js
module.exports = require('@haul-bundler/core/dist/webpack/plugins/WebpackBasicBundlePlugin');
```

File: packages/haul-basic-bundle-webpack-plugin/package.json [hunk 4]
```diff
@@ -1,21 +0,0 @@
...
-  "author": "Pawel Trysla <pawel.trysla@callstack.com>",
-  "repository": "github:callstack/haul",
-  "license": "MIT",
-  "bugs": "https://github.com/callstack/haul/issues",
-  "peerDependencies": {
-    "webpack": "4.x",
-    "webpack-sources": "1.x"
-  },
-  "dependencies": {
-    "concat-with-sourcemaps": "^1.1.0"
```
Comment (zamotany) [inline packages/haul-basic-bundle-webpack-plugin/package.json:18]: This dependency can be removed.

File: packages/haul-basic-bundle-webpack-plugin/package.json [hunk 5]
```diff
@@ -1,21 +0,0 @@
...
-  "files": [
-    "build/"
-  ],
-  "author": "Pawel Trysla <pawel.trysla@callstack.com>",
-  "repository": "github:callstack/haul",
-  "license": "MIT",
-  "bugs": "https://github.com/callstack/haul/issues",
-  "peerDependencies": {
-    "webpack": "4.x",
-    "webpack-sources": "1.x"
```
Comment (zamotany) [inline packages/haul-basic-bundle-webpack-plugin/package.json:15]: Those dependencies can be removed and replaced with:
```
"@haul-bundler/core": ">=0.19.0"
```

File: packages/haul-core/src/preset/makeConfigFactory.ts (hunk 1 above)
Comment (zamotany) [reply packages/haul-core/src/preset/makeConfigFactory.ts:29]: `BundleWebpackBasicPlugin` IMO doesn't make sense `BasicBundle` should go together always, so I'd suggest `BasicBundleWebpackPlugin`

Comment (NoemiRozpara) [reply packages/haul-core/src/preset/makeConfigFactory.ts]: Yeah, I also noticed that, changed all occurrences to `BasicBundleWebpackPlugin`

File: packages/haul-basic-bundle-webpack-plugin/README.md [hunk 6]
```diff
@@ -1,42 +0,0 @@
...
-[license]: https://opensource.org/licenses/MIT
-[prs-welcome-badge]: https://img.shields.io/badge/PRs-welcome-brightgreen.svg?style=flat-square
-[prs-welcome]: http://makeapullrequest.com
-[coc-badge]: https://img.shields.io/badge/code%20of-conduct-ff69b4.svg?style=flat-square
-[coc]: https://github.com/callstack/haul/blob/master/CODE_OF_CONDUCT.md
-[chat-badge]: https://img.shields.io/badge/chat-discord-brightgreen.svg?style=flat-square&colorB=7289DA&logo=discord
-[chat]: https://discord.gg/zwR2Cdh
-
-[version]: https://img.shields.io/npm/v/@haul-bundler/basic-bundle-webpack-plugin.svg?style=flat-square
-[package]: https://www.npmjs.com/package/@haul-bundler/basic-bundle-webpack-plugin
```
Comment (zamotany) [inline packages/haul-basic-bundle-webpack-plugin/README.md:42]: Can you restore this file with the following content:
```md
# @haul-bundler/basic-bundle-webpack-plugin

`BasicBundleWebpackPlugin` was moved into `@haul-bundler/core` package. This module exists only for backwards compatibility and should not be used.
```

File: packages/haul-core/src/preset/makeConfigFactory.ts (hunk 1 above)
Comment (thymikee) [reply packages/haul-core/src/preset/makeConfigFactory.ts]: Lol, a typo. I meant `BasicBundleWebpackPlugin` of course

**Pull request merged**